### PR TITLE
Remove perPage. page and q param and switch to full query string being rql

### PR DESCRIPTION
**Pull request by hairmare**
* [x] remove ``q``, ``perPage`` and ``page`` params from tests and link headers
* [x] add some docs explaining what is going on with ``,`` vs ``%2C`` in Link headers
* [x] update ``composer.lock`` with a tagged version of rql-parser-bundle when https://github.com/libgraviton/GravitonRqlParserBundle/pull/25 is merged in a tagged release.

I'm aiming at tagging a new graviton release as soon as this has been merged...

### CODE REVIEW

File: src/Graviton/CoreBundle/Tests/Controller/ModuleControllerTest.php [hunk 1]
```diff
@@ -52,24 +52,24 @@ public function setUp()
...
-            '<http://localhost/core/module?page=2&perPage=1>; rel="next"',
-            explode(',', $response->headers->get('Link'))
+            '<http://localhost/core/module?limit(1%2C1)>; rel="next"',
+            $response->headers->get('Link')
         );
 
         $this->assertContains(
-            '<http://localhost/core/module?page=5&perPage=1>; rel="last"',
-            explode(',', $response->headers->get('Link'))
+            '<http://localhost/core/module?limit(1%2C4)>; rel="last"',
```
Comment (lapistano) [inline src/Graviton/CoreBundle/Tests/Controller/ModuleControllerTest.php]: so the offset (aka 'page') is now defined by the 2nd integer?  here: present me the result 1 to 4

File: src/Graviton/CoreBundle/Tests/Controller/AppControllerTest.php [hunk 2]
```diff
@@ -88,32 +88,32 @@ public function testFindAll()
...
-            '<http://localhost/core/app?q=eq(showInMenu%2Ctrue)&page=2&perPage=1>; rel="last"',
+            '<http://localhost/core/app?eq(showInMenu%2Ctrue)&limit(1%2C1)>; rel="last"',
             $response->headers->get('Link')
         );
 
     }
 
     /**
-     * rql limit() should be *never* overwritten by $_GET['perPage'] *or* default value
+     * rql limit() should *never* by overwritten by default value
```
Comment (lapistano) [inline src/Graviton/CoreBundle/Tests/Controller/AppControllerTest.php]: rql limit() should *never* _be_ overwritten by default value

File: src/Graviton/CoreBundle/Tests/Controller/ModuleControllerTest.php (hunk 1 above)
Comment (hairmare) [inline src/Graviton/CoreBundle/Tests/Controller/ModuleControllerTest.php]: limit(1,4) == return 1 item starting from item 4.

File: src/Graviton/CoreBundle/Tests/Controller/AppControllerTest.php [hunk 3]
```diff
@@ -121,45 +121,40 @@ public function testGetAppPagingWithRql()
...
         $this->assertContains(
-            '<http://localhost/core/app?q=limit(1)&page=2&perPage=1>; rel="last"',
+            '<http://localhost/core/app?limit(1%2C1)>; rel="last"',
             $response->headers->get('Link')
         );
 
         // "page" override - rql before get
         $client = static::createRestClient();
-        $client->request('GET', '/core/app?perPage=2&page=1&q=limit(1,1)');
+        $client->request('GET', '/core/app?limit(1,1)');
```
Comment (lapistano) [inline src/Graviton/CoreBundle/Tests/Controller/AppControllerTest.php]: If I understood correctly `/core/app?perPage=2&page=1&q=limit(1,1)` should now be `/core/app?limit(1,2)`, no?

File: src/Graviton/CoreBundle/Tests/Controller/AppControllerTest.php [hunk 4]
```diff
@@ -121,45 +121,40 @@ public function testGetAppPagingWithRql()
...
 
         $this->assertContains(
-            '<http://localhost/core/app?q=limit(1)>; rel="self"',
+            '<http://localhost/core/app?limit(1)>; rel="self"',
             $response->headers->get('Link')
         );
 
         $this->assertContains(
-            '<http://localhost/core/app?q=limit(1)&page=2&perPage=1>; rel="next"',
+            '<http://localhost/core/app?limit(1%2C1)>; rel="next"',
```
Comment (lapistano) [inline src/Graviton/CoreBundle/Tests/Controller/AppControllerTest.php]: same here (see https://github.com/libgraviton/graviton/pull/218/files#r36723552)

File: src/Graviton/CoreBundle/Tests/Controller/AppControllerTest.php [hunk 5]
```diff
@@ -88,32 +88,32 @@ public function testFindAll()
...
 
         $this->assertContains(
-            '<http://localhost/core/app?q=eq(showInMenu%2Ctrue)>; rel="self"',
+            '<http://localhost/core/app?eq(showInMenu%2Ctrue)&limit(1)>; rel="self"',
             $response->headers->get('Link')
         );
 
         $this->assertContains(
-            '<http://localhost/core/app?q=eq(showInMenu%2Ctrue)&page=2&perPage=1>; rel="next"',
+            '<http://localhost/core/app?eq(showInMenu%2Ctrue)&limit(1%2C1)>; rel="next"',
```
Comment (lapistano) [inline src/Graviton/CoreBundle/Tests/Controller/AppControllerTest.php]: same here (see https://github.com/libgraviton/graviton/pull/218/files#r36723552)

File: src/Graviton/CoreBundle/Tests/Controller/AppControllerTest.php [hunk 6]
```diff
@@ -121,45 +121,40 @@ public function testGetAppPagingWithRql()
...
 
         $this->assertContains(
-            '<http://localhost/core/app?q=limit(1%2C1)>; rel="self"',
+            '<http://localhost/core/app?limit(1%2C1)>; rel="self"',
             $response->headers->get('Link')
         );
 
         // we're passing page=1, but should be on last.. so next and last should be identical
-        $nextAndLastUrl = 'http://localhost/core/app?q=limit(1%2C1)&page=2&perPage=1';
+        $nextAndLastUrl = 'http://localhost/core/app?limit(1%2C1)';
```
Comment (lapistano) [inline src/Graviton/CoreBundle/Tests/Controller/AppControllerTest.php]: same here (see https://github.com/libgraviton/graviton/pull/218/files#r36723552)

File: src/Graviton/CoreBundle/Tests/Controller/AppControllerTest.php [hunk 7]
```diff
@@ -121,45 +121,40 @@ public function testGetAppPagingWithRql()
...
 
         $this->assertContains(
-            '<http://localhost/core/app?q=limit(1)&page=2&perPage=1>; rel="next"',
+            '<http://localhost/core/app?limit(1%2C1)>; rel="next"',
             $response->headers->get('Link')
         );
 
         $this->assertContains(
-            '<http://localhost/core/app?q=limit(1)&page=2&perPage=1>; rel="last"',
+            '<http://localhost/core/app?limit(1%2C1)>; rel="last"',
```
Comment (lapistano) [inline src/Graviton/CoreBundle/Tests/Controller/AppControllerTest.php]: If I understood correctly `/core/app?perPage=1&page=2` should now be `/core/app?limit(1,2)`, no?

File: src/Graviton/CoreBundle/Tests/Controller/AppControllerTest.php [hunk 8]
```diff
@@ -88,32 +88,32 @@ public function testFindAll()
...
 
         $this->assertContains(
-            '<http://localhost/core/app?q=eq(showInMenu%2Ctrue)&page=2&perPage=1>; rel="next"',
+            '<http://localhost/core/app?eq(showInMenu%2Ctrue)&limit(1%2C1)>; rel="next"',
             $response->headers->get('Link')
         );
 
         $this->assertContains(
-            '<http://localhost/core/app?q=eq(showInMenu%2Ctrue)&page=2&perPage=1>; rel="last"',
+            '<http://localhost/core/app?eq(showInMenu%2Ctrue)&limit(1%2C1)>; rel="last"',
```
Comment (lapistano) [inline src/Graviton/CoreBundle/Tests/Controller/AppControllerTest.php]: same here (see https://github.com/libgraviton/graviton/pull/218/files#r36724022)

File: src/Graviton/CoreBundle/Tests/Controller/AppControllerTest.php (hunk 7 above)
Comment (hairmare) [inline src/Graviton/CoreBundle/Tests/Controller/AppControllerTest.php]: First page is ``limit(1,0)`` so having 2 pages the second one is ``limit(1,1)``

File: src/Graviton/CoreBundle/Tests/Controller/AppControllerTest.php (hunk 3 above)
Comment (hairmare) [inline src/Graviton/CoreBundle/Tests/Controller/AppControllerTest.php]: The ``perPage=2`` in the original example was rather senseless. I updated the complete example to reflect what it should have been all along. Right now it doesn't request a non full page like before but reflects what we want to test more clearly.

Most of what I fixed in ``testGetAppPagingWithRql `` is right after this note...

File: src/Graviton/CoreBundle/Tests/Controller/AppControllerTest.php (hunk 4 above)
Comment (hairmare) [inline src/Graviton/CoreBundle/Tests/Controller/AppControllerTest.php]: ``perPage`` was silly as explained.. This requests the first page of 2 records, ``next`` now references the second record.

File: src/Graviton/CoreBundle/Tests/Controller/AppControllerTest.php (hunk 3 above)
Comment (hairmare) [inline src/Graviton/CoreBundle/Tests/Controller/AppControllerTest.php]: Also lots of this tested that rql limit overrides the setting from perPage, that's another reason it looks funky in the PR.

**Pull request merged**